autogluon/utils/tabular/ml/models/knn/knn_model.py

```diff
@@ -28,9 +27,8 @@ def _get_model_type(self):
             return KNeighborsClassifier
 
     def preprocess(self, X):
-        cat_columns = X.select_dtypes(['category']).columns
-        X = X.drop(cat_columns, axis=1)  # TODO: Test if crash when all columns are categorical
         X = super().preprocess(X).fillna(0)
+        X = X.to_numpy(dtype=np.float32)
```

Comment:
What happens if `X` originally only contained `int` types say?

Reply:
They would be converted to float32 and the accuracy/predictions would be the same. However, if they were all uint8, it would be slightly less efficient to convert them. This is a level of optimization that we could consider in future, but would have to be tested.